### PR TITLE
Change Mdns deployment type to StatefulSet

### PR DESCRIPTION
**Pull request by omersch381**
This commit changes the Mdns deployment type as well
as updating how the predictable IP map is generated.

### CODE REVIEW

**Comment by omersch381** [comment]
I keep getting
error: error creating buildah builder: initializing source docker://registry.build06.ci.openshift.org/ci/managed-clonerefs@sha256:8045e4633e3e825a92fa4fbe58ad6e7cc73bd45e5c0f207a0cc0c3666b49949f: pinging container registry registry.build06.ci.openshift.org: Get "https://registry.build06.ci.openshift.org/v2/": dial tcp: lookup registry.build06.ci.openshift.org on 172.30.0.10:53: no such host

File: api/v1beta1/designatemdns_types.go [hunk 1]
```diff
@@ -45,6 +45,13 @@ type DesignateMdnsSpecBase struct {
 	// Common input parameters for all Designate services
 	DesignateTemplate `json:",inline"`
 
+    // +kubebuilder:validation:Optional
```
Comment (beagles) [inline api/v1beta1/designatemdns_types.go:48]: Inconsistent indentation maybe?

**Comment by beagles** [comment]
Small nit  but looks good otherwise.

File: api/v1beta1/designatemdns_types.go (hunk 1 above)
Comment (omersch381) [reply api/v1beta1/designatemdns_types.go]: Thanks, done.

File: controllers/designate_controller.go [hunk 2]
```diff
@@ -762,25 +761,14 @@ func (r *DesignateReconciler) reconcileNormal(ctx context.Context, instance *des
...
-	nodes, err := helper.GetKClient().CoreV1().Nodes().List(ctx, metav1.ListOptions{})
-	if err != nil {
-		return ctrl.Result{}, err
-	}
-
-	var nodeNames []string
-	for _, node := range nodes.Items {
-		nodeNames = append(nodeNames, fmt.Sprintf("mdns_%s", node.Name))
+	// Handle Mdns predictable IPs configmap
+	mdnsReplicaCount := int(instance.Spec.DesignateMdns.Replicas)
```
Comment (beagles) [inline controllers/designate_controller.go:765]: should be

`mdnsReplicaCount := int(i*nstance.Spec.DesignateMdns.Replicas)`

File: controllers/designatemdns_controller.go [hunk 3]
```diff
@@ -641,45 +627,41 @@ func (r *DesignateMdnsReconciler) reconcileNormal(ctx context.Context, instance
 			return ctrl.Result{RequeueAfter: time.Duration(1) * time.Second}, nil
 		}
 
-		if instance.Status.ReadyCount == instance.Status.DesiredNumberScheduled {
+		if instance.Status.ReadyCount == instance.Spec.Replicas {
```
Comment (beagles) [inline controllers/designatemdns_controller.go:630]: should be
` if instance.Status.ReadyCount == *instance.Spec.Replicas {`

File: controllers/designate_controller.go (hunk 2 above)
Comment (omersch381) [reply controllers/designate_controller.go]: Thanks, I have just fixed it.

File: controllers/designatemdns_controller.go (hunk 3 above)
Comment (omersch381) [reply controllers/designatemdns_controller.go]: Thanks, I have just fixed it.

**Comment by beagles** [comment]
So cool, the new checks  caught a problem. Seems we cannot remove status fields.

**Comment by omersch381** [comment]
> So cool, the new checks caught a problem. Seems we cannot remove status fields.

That's weird, because it didn't fail the pre commit checks on my local environment, and as far as I understand, we are now bound to have fields that we won't use...

**Pull request merged**